File: backend/Api/Models/Driver/Driver.cs
using System.Text.Json.Serialization;
using Api.Core.Models;

namespace Api.Models.Driver
{
    public class Driver : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string? LicenseNumber { get; set; }
        public string Vehicle { get; set; } = string.Empty;
        public string? VehicleRegistration { get; set; }
        public string Status { get; set; } = "Available"; // Available, On Delivery, Off Duty
        public decimal Rating { get; set; } = 0;
        public bool IsActive { get; set; } = true;

        // Navigation properties
        [JsonIgnore]
        public virtual ICollection<Delivery.Delivery> Deliveries { get; set; } = new List<Delivery.Delivery>();
    }
}

File: frontend/src/app/core/domain/driver/driver.ts
/**
 * Represents a Driver
 */
export interface Driver {
  /** ID */
  id: number;

  /** Name - driver name */
  name: string;

  /** Phone - contact phone */
  phone: string;

  /** License Number - driver's license */
  licenseNumber?: string | null;

  /** Vehicle - vehicle description */
  vehicle: string;

  /** Vehicle Registration - license plate */
  vehicleRegistration?: string | null;

  /** Status - driver status */
  status?: string; // Available, On Delivery, Off Duty

  /** Rating - driver rating (0-5) */
  rating?: number;

  /** Is Active - is the driver active */
  isActive?: boolean;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}

/**
 * Driver Status enum
 */
export enum DriverStatus {
  Available = 'Available',
  OnDelivery = 'On Delivery',
  OffDuty = 'Off Duty',
}
